gm4_trash_cans/data/gm4_trash_cans/functions/check_dropper.mcfunction

```diff
@@ -0,0 +1,17 @@
+# @s = trash_can
+# run from update
+
+# Visuals
+data merge entity @s {CustomName:'"gm4_trash_can"',ArmorItems:[{},{},{},{id:"minecraft:magma_block",Count:1b,tag:{CustomModelData:1}}]}

```

Comment:
This function seems to run every clock pulse when there is one of @s around.
I see no reason for running this expensive merge operation on every pulse. Should this be in create.mcfuncrion?

Reply:
Whenever water touches the side of the Trash Can it cools down and isn't able to burn the items inside.
(This is how players can disable it)
When it is "cooled" down it turns the magma block into a cobblestone block to help show the player it is disabled. I will try and find a way to make it less of a problem.